File: apps/api/src/main/java/to/bconnect/api/storage/Address.java
package to.bconnect.api.storage;

import jakarta.persistence.Embeddable;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Embeddable
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class Address {
    @NotBlank
    @Size(max = 5)
    private String zipcode;

    @NotNull
    @Enumerated(EnumType.STRING)
    private Region state;

    @NotBlank
    private String city;

    @NotBlank
    private String street;

    private String detail;

    @NotNull
    private BigDecimal latitude;

    @NotNull
    private BigDecimal longitude;
}

File: packages/mocks/src/overrides/_address.ts
import type { Address, Region } from '@bconnect/api-client'

/** mock 주소. 화면은 state(지역 필터)·city(표시)·street 만 읽는다. 위경도는 config/address 와 같은 placeholder 0. */
export function addressOf(
  state: Region,
  city: string,
  street = '',
  detail?: string,
  zipcode = '00000'
): Address {
  return {
    zipcode,
    state,
    city,
    street,
    detail,
    latitude: 0,
    longitude: 0,
  }
}
